server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -1515,14 +1510,7 @@ public Builder removeIndexTemplate(String name) {
         }
 
         public DataStream dataStream(String dataStreamName) {
-            previousIndicesLookup = null;
-
-            DataStreamMetadata dataStreamMetadata = (DataStreamMetadata) customs.get(DataStreamMetadata.TYPE);
-            if (dataStreamMetadata != null) {
-                return dataStreamMetadata.dataStreams().get(dataStreamName);
-            } else {
-                return null;
-            }
+            return dataStreams().dataStreams().get(dataStreamName);
         }
 
         public Builder dataStreams(Map<String, DataStream> dataStreams, Map<String, DataStreamAlias> dataStreamAliases) {
```

Comment:
In light of my previous comment, I'm curious whether this is better with the current signature versus perhaps being named `dataStreamMetadata` and receiving a `DataStreamMetadata`.

Reply:
Yea lets do a follow-up for this IMO. Looking over this all these methods are needlessly complex and we can save tons of code here doing similar things like I did here already (as well as make the code a little more obviously correct in the process :)).